### DISCUSSION TITLE
Create UBQ bonding curve to burn UBQ and mint NFTs

### DISCUSSION THREAD

**Discussion by sergfeldman** [pavlovcik/uad-ui#428] [Protocol Specifications]
- [ ] Create the UBQ bonding curve smart contract (fork is preferred)
- [ ] Update UI component to allow for interaction with this smart contract.
- [ ] Implement the "buy with any token" flow (powered by Uniswap.)

---
Overview:

1. Get the tokens that the user has
2. Exchange to UBQ
3. Send UBQ to bonding curve contract (this should be recoverable by the admin wallet.)
4. Mint UbiquiStick NFT
5. Transfer the UbiquiStick NFT to the buyer

Note:

- Please use our standard access control pattern from the UbiquityManager e.g. `UBQ_MINTER_ROLE` to allow the admin wallet to collect the raised UBQ

**Comment by codeawayaks** [comment]
You can assign this to me.

**Comment by sergfeldman** [comment]
@codeawayaks We offer solving bounties one at a time to minimize blocking other developers to move things forward. Which one issue you want to work on first?

**Comment by sergfeldman** [comment]
> @sergfeldman I am very interested about this task. But I need your explain a bit more.

@pufferdev0807 Please, post your clarifying questions here.
We will make a decision on assigning you after you ask questions.

**Comment by zgorizzo69** [comment]
hello @sergfeldman I am very familiar with bonding curve (bancor formula) as you can see here https://github.com/meTokens/meTokens-core
and I get what you want to do here :smile: 
 let me know if I can work on it

**Comment by Draeieg** [comment]
@zgorizzo69 Already did most of the mathematical modeling for this just need to refine a few parameters https://www.desmos.com/calculator/o311zdypy1, we can hop on a call for more details if the decision is to have you working on this

**Reply by 0x4007** [reply]
You should write this out as a spec because we now have a Bounty Hunter implementing this.

**Comment by zgorizzo69** [comment]
> @zgorizzo69 Already did most of the mathematical modeling for this just need to refine a few parameters https://www.desmos.com/calculator/o311zdypy1, we can hop on a call for more details if the decision is to have you working on this

Sure ! Let's wait for @sergfeldman @pavlovcik decision then

**Comment by 0x4007** [comment]
> hello @sergfeldman I am very familiar with bonding curve (bancor formula) as you can see here https://github.com/meTokens/meTokens-core and I get what you want to do here 😄 let me know if I can work on it

That's dope but I feel like this particular bounty is quite low in our priority list. Given your experience with this project and Solidity expertise, your contributions will be more impactful if we have you focus on the more important issues you have also expressed interest in.

**Comment by sergfeldman** [comment]
Note for the bounty hunters.

This issue should be divided to smaller tasks with the specific deliverables.
For example, 
1. Update the UI component.
2. Implement the logic "Buy with any token".

Please, suggest smaller tasks when apply for this bounty.

**Comment by 0x4007** [comment]
> This issue should be divided to smaller tasks with the specific deliverables. For example,
> 
> 1. Update the UI component.
> 2. Implement the logic "Buy with any token".
> 
> Please, suggest smaller tasks when apply for this bounty.

I encourage you to break up the issue into separate issues. It should be on us to do that.

**Comment by sergfeldman** [comment]
Created 2 issues
https://github.com/ubiquity/ubiquity-dollar/issues/429
https://github.com/ubiquity/ubiquity-dollar/issues/430

@pavlovcik @0xcodercrane Please, advise whether it makes sense to split #430 into 2-3 issues

**Reply by 0x4007** [reply]
I think #429 should have more details, ideally like a mockup or wireframe.

**Reply by 0x4007** [reply]
I think splitting #430 into a "mvp" and "final" versions make sense.

- mvp should directly accept UBQ only
- final should incorporate something like Uniswap for added convenience to start from any token.

**Comment by 0x4007** [comment]
Perhaps we should remove the pricing labels for tasks like this (now they have subtasks with separate price labels) and leave these for reference.

---

Update: I just realized this is under Dscussions so it should be fine.
